analysis/measure_definition.py

```diff
@@ -171,14 +183,21 @@ def num_months(from_, to_):
 elif 'hba1c_diab' in args.test:
     denominator = denominator & is_diabetic
 
-# For mean, sum(numeric_value) / sum(patients who had a test) = mean value of tests (ratio column)
-if 'mean' in args.test:
-    numerator = numeric_value
-    denominator = denominator & has_codelist_event
-elif 'ref' in args.test: 
-    numerator = tests_outside_ref
+# Remove tests with unreliable numeric values for measures that depend on that field
+if ('mean' in args.test) | ('ref' in args.test):
+    has_codelist_event = (codelist_events.where(
+                            (codelist_events.numeric_value.is_not_null()) & 
+                            (codelist_events.numeric_value >= 0))
```

Comment:
I think this isn't excluding zeros as intended?
```suggestion
                            (codelist_events.numeric_value > 0))
```